R/unionHUCSet.R

```diff
@@ -0,0 +1,47 @@
+#' Union HUC Set
+#' 
+#' Aggregates geometry for a large collection of HUCs. Note that subhucPoly must be all potential HUCs required.
+#' 
+#' @param aggrHUCs The list of aggregate HUCs to be processed per HUC_aggregator
+#' @param fromHUCs The list of 'fromHUCs' to be processed per fromHUC_finder
+#' @param subhucPoly A subset of HUCs for the region in questioned.
+#' @return The subhucPoly data frame with requested aggregated HUCs unioned together. 
+#' Note that Polygon 'ID's are set to the HUC id and areas are added up appropriately. 
+#' @author David Blodgett \email{dblodgett@@usgs.gov}
+#' @importFrom maptools unionSpatialPolygons
+#' @export
+#' @examples
+#' \dontrun{
+#' load(system.file("extdata","testhucpoly.rda",package="HUCAgg"))
+#' plot(testhucPoly)
+#' hucList<-testhucPoly@data$HUC12
+#' fromHUC<-sapply(hucList,fromHUC_finder,hucs=testhucPoly@data$HUC12,tohucs=testhucPoly@data$TOHUC)
+#' aggrHUCs<-sapply(hucList, HUC_aggregator, fromHUC=fromHUC)
+#' testhucPoly<-unionHUCSet(aggrHUCs, fromHUC, testhucPoly)
+#' plot(subset(testhucPoly,testhucPoly@data$HUC12 %in% "070900020904"), add=TRUE, col=rgb(1,0,0,.3))
+#' plot(subset(testhucPoly,testhucPoly@data$HUC12 %in% "070900020702"), add=TRUE, col=rgb(1,1,0,.3))
+#' }
+unionHUCSet<-function(aggrHUCs,fromHUCs,subhucPoly) {
+  upstream_size<-sapply(aggrHUCs, length)
+  for ( setSize in 1:max(upstream_size)) {
+    hucs<-names(upstream_size[which(upstream_size==setSize)])
+    for ( huc in hucs ) {
+      fromHUC_local<-c(unlist(fromHUCs[huc][[1]]))
+      if(length(fromHUC_local)>50) {print(paste(huc,'has',length(fromHUC_local),'contributing hucs'))}
```

Comment:
do you want `message` instead of print for these? And an ability to set verbose=TRUE to get them vs hush them? 

Also, what is the significance of 50, and why does >50 get a message/print and below that does not? >50 takes along time? If so, you might want to include that in the message